### ISSUE TITLE
Squiz.PHP.NonExecutableCode.Unreachable when using ternary operator

### ISSUE THREAD

**Issue by kkmuffme** [squizlabs/PHP_CodeSniffer#2857]
Getting Squiz.PHP.NonExecutableCode.Unreachable for following (valid) code:

```php
$var = 'hello';
echo $var == 'foo' ? 'bar' : die( 'world' );

echo 'example';
```

**Comment by gsherwood** [comment]
This is absolutely a bug, but I've been trying to decide if I remove all these non-code-style sniffs in version 4 instead of trying to support them. It's not what PHPCS is good at and not a collection of sniffs I have time to keep updated.

I'll leave this open as a bug until I decide either way, but no work will be done on it until I do.

**Comment by kkmuffme** [comment]
Overall, these work really good and there's few bugs in those, so I think it would be great if they could be kept, as they are really useful.

**Comment by dpi** [comment]
Affects external issue https://www.drupal.org/project/date_recur/issues/3250423

**Comment by jrfnl** [comment]
FYI: the fix for this issue is included in today's [PHP_CodeSniffer 3.8.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.8.0).

As per #3932, development on PHP_CodeSniffer will continue in the [PHPCSStandards/PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer) repository. If you want to stay informed, you may want to start "watching" that repo (or watching releases from that repo).
